frontend_tests/node_tests/stream_topic_history.js

```diff
@@ -65,6 +93,7 @@ run_test("basics", () => {
         stream_id,
```

Comment:
The code changes generally look reasonable, but I don't understand what you did with the tests.  The comment "Test that duplicate remove does not crash us." seems no longer correct, for example.

Reply:
Posted a comment that I think would resolve my confusion, above.